core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2334,3 +2335,26 @@ strip eq arr@Array{..} = do
             then getLast prev low
             else return cur
         else return cur
+
+-- | Bubble sorts the elements.
+--
+-- /Pre-release/
+{-# INLINE bubbleAsc #-}
+bubbleAsc :: (MonadIO m, Storable a) =>
+    (a -> a -> Ordering) -> Array a -> m ()
+bubbleAsc cmp0 arr = go (l-2)
+
+    where
+
+    l = length arr
+
+    go i =
+        when (i >= 0) $ do
+            x1 <- getIndexUnsafe i arr
+            x2 <- getIndexUnsafe (i+1) arr
+            case x1 `cmp0` x2 of
+                GT -> do
+                    _ <- putIndexUnsafe i x2 arr
+                    _ <- putIndexUnsafe (i+1) x1 arr
+                    go (i - 1)
```

Comment:
We can pass `x2` to `go` so that we do not have to read it again in the next iteration of the loop.